File: src/codingProblems/Python/two_sum.py
nums = [0, 11, 9, 3]
target = 9


class Solution(object):

    def __init__(self, nums, target):
        self.nums = nums
        self.target = target

    def two_sum(self, nums, target):

        arr = {}

        for i, value in enumerate(nums):

            diff = target - value

            if diff in arr:
                return [arr[diff], i]

            arr[value] = i

        return [-1, -1]


res = Solution(nums, target)

print(res.two_sum(nums, target))

File: src/codingProblems/JavaScript/TwoSum.js
const nums = [5, 6, 1, 4];
const target = 5;

console.log(twoSum(nums, target));


/*
    HashMap - O(n)
*/
// function twoSum(nums, target) {

//     let res;
//     const map = new Map();

//     for (let i = 0; i < nums.length; i++) {

//         if (map.has(target - nums[i])) {
//             return res = [i, map.get(target - nums[i])];
//         } else {
//             map.set(nums[i], i);
//         }
//     }
// }

/*
    Nested for loops - O(n^2)
*/
function twoSum(nums, target) {

    let res;

    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (target - nums[j] == nums[i]) {
                return res = [i, j];
            }
        }
    }
    return null;
}